pywhat/distribution.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+from typing import Optional
+
+from pywhat.helper import AvailableTags, InvalidTag
+
+
+class Distribution:
+    """
+    A distribution is an object containing the regex
+    But the regex has gone through a filter process.
+
+    Example filters:
+    * {"Tags": ["Networking"]}
+    * {"Tags": ["Identifiers"], "ExcludeTags": ["Credentials"], "MinRarity": 0.6}
+    """
+
+    def __init__(self, filters_dict: Optional[dict] = None):
+        tags = AvailableTags().get_tags()
+        self._dict = dict()
+        if filters_dict is None:
+            filters_dict = dict()
+
+        self._dict["Tags"] = set(filters_dict.setdefault("Tags", tags))
+        self._dict["ExcludeTags"] = set(filters_dict.setdefault("ExcludeTags", set()))
+        self._dict["MinRarity"] = filters_dict.setdefault("MinRarity", 0)
+        self._dict["MaxRarity"] = filters_dict.setdefault("MaxRarity", 1)
+
+        if not self._dict["Tags"].issubset(tags) or not self._dict["ExcludeTags"].issubset(tags):
+            raise InvalidTag("Passed filter contains tags that are not used by 'what'")
+
+        self._load_regexes()
+
+    def _load_regexes(self):
+        path = "Data/regex.json"
+        fullpath = os.path.join(os.path.dirname(os.path.abspath(__file__)), path)
+        with open(fullpath, "r", encoding="utf-8") as myfile:
+            self._regexes = json.load(myfile)
+        self._filter()
+
+    def _filter(self):
+        temp_regexes = []
+        min_rarity = self._dict["MinRarity"]
+        max_rarity = self._dict["MaxRarity"]
+        for regex in self._regexes:
+            if (
+                min_rarity <= regex["Rarity"] <= max_rarity
+                and set(regex["Tags"]) & self._dict["Tags"]
+                and not set(regex["Tags"]) & self._dict["ExcludeTags"]
+            ):
+                temp_regexes.append(regex)
+
+        self._regexes = temp_regexes
+
+    def get_regexes(self):
+        return list(self._regexes)
+
+    def get_filter(self):
+        return dict(self._dict)
+
+    def __repr__(self):
+        return f"Distribution({self._dict})"
+
+    def __and__(self, other):

```

Comment:
I would have gone with `__add__` here,  but I suppose it makes more natural sense to use `__and__`

Reply:
Yeah, I have been thinking about __add__, but its behaviour with rarities would be unclear, so I used bitwise operators instead.